### PR TITLE
Working dir fields

### PR DESCRIPTION
**Pull request by iGroovyboy**
Hi! This PR aimed to add fields (with default values of "." and "/application") to the generator form to allow user set custom docker volume paths. This changes these paths in all corresponding docker partials (like mysql and so on), nginx config and docs.
It looks like [this](https://imgur.com/a/ptF9Dqb). 
If you consider this change is ok in general, then please, take a look at the concerns i have:

- fields labels: naming are basically taken from docker documentation but maybe i should add some inline notes or some sort of tooltips to give some extra info
- inline styles [here](https://github.com/phpdocker-io/phpdocker.io/compare/master...iGroovyboy:working-dir-fields?expand=1#diff-2c763c5544807ec9a88a659c6834d3b98bc34298904d98d4f97195deb5f10483R13) look a bit nasty, those were added to add padding to 'Note'
- hasWorkingDirOptions() returns true. This looks a bit hacky. ([look here](https://github.com/phpdocker-io/phpdocker.io/compare/master...iGroovyboy:working-dir-fields?expand=1#diff-830576e68b0bb113912483701458e25576dfe7fea4721f20c4fc86bda787498bR175))

If you have better ideas on how to deal with those up above of my concerns or anything else, please let me know.

### CODE REVIEW

**Comment by luispabon** [comment]
Thank you for your contribution and your pull request 👍🏽 

I refactored yesterday the generator in a way that broke your pull request (docker-compose file is no longer twig-templated and other bits and pieces) so I took the liberty of pushing an update into your branch addressing the changes. They seem to work OK. I'll review your changes next.

File: src/Form/Generator/WorkingDirType.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+declare(strict_types=1);
+
+
```
Comment (luispabon) [inline src/Form/Generator/WorkingDirType.php:4]: Please remove extra newline.

File: src/Form/Generator/WorkingDirType.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?php
+declare(strict_types=1);
+
+
+namespace App\Form\Generator;
+
+
```
Comment (luispabon) [inline src/Form/Generator/WorkingDirType.php:7]: Please remove extra newline.

File: src/Form/Generator/WorkingDirType.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                'label'       => 'Source volume',
+                'data'        => '.',
+            ])
+            ->add('dockerWorkingDir', TextType::class, [
+                'label'       => 'Target volume',
+                'data'        => '/application',
+            ]);
+    }
+
+}
```
Comment (luispabon) [inline src/Form/Generator/WorkingDirType.php:26]: Please add newline at the end of the file.

File: src/PHPDocker/Project/ServiceOptions/WorkingDir.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<?php
+declare(strict_types=1);
+
+
```
Comment (luispabon) [inline src/PHPDocker/Project/ServiceOptions/WorkingDir.php:4]: Please remove extra newline.

File: src/PHPDocker/Project/ServiceOptions/WorkingDir.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+namespace App\PHPDocker\Project\ServiceOptions;
+
+
```
Comment (luispabon) [inline src/PHPDocker/Project/ServiceOptions/WorkingDir.php:7]: Please remove extra newline.

File: src/PHPDocker/Project/ServiceOptions/WorkingDir.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        return $this->dockerWorkingDir;
+    }
+
+    public function setDockerWorkingDir(string $dockerWorkingDir = '/application'): self
+    {
+        $this->dockerWorkingDir = $dockerWorkingDir;
+
+        return $this;
+    }
+}
```
Comment (luispabon) [inline src/PHPDocker/Project/ServiceOptions/WorkingDir.php:38]: Please add newline at the end of the file.

File: src/Controller/GeneratorController.php [hunk 7]
```diff
@@ -91,11 +92,18 @@ private function hydrateProject(array $formData): Project
         $appData    = $formData['applicationOptions'];
         $appOptions = new Application(applicationType: $appData['applicationType'], uploadSize: $appData['uploadSize']);
 
+        $workingDirData    = $formData['workingDirOptions'];
+        $workingDirOptions = new WorkingDir(
+            localWorkingDir: $workingDirData['localWorkingDir'],
+            dockerWorkingDir: $workingDirData['dockerWorkingDir']
```
Comment (luispabon) [inline src/Controller/GeneratorController.php:98]: Did you check what happens when users add a trailing slash at the end of those two values in the form? We might possibly need to strip any trailing slashes (eg using `rtrim($value, '/')`) here to make sure we don't end up later on with invalid paths on the generated project.

File: src/PHPDocker/Template/nginx.conf.twig [hunk 8]
```diff
@@ -13,7 +13,7 @@ server {
             {% set indexFile = 'index' %}
         {% endif %}
 
-    root /application/web;
+    root {{ dockerWorkingDir }}/web;
```
Comment (luispabon) [inline src/PHPDocker/Template/nginx.conf.twig:16]: See comment above on the controller about form values with trailing slashes, they could potentially cause issues here and below in `/public`

File: templates/generator.html.twig [hunk 9]
```diff
@@ -10,6 +10,12 @@
 {% set halfColumn = 'col col-lg-5 col-lg-offset-1 col-md-5 col-md-offset-1 col-sm-12 col-xs-12' %}
 
 {% block body %}
+    <style>
+        .pl-30  {
+            padding-left: 30px;
+        }
+    </style>
```
Comment (luispabon) [inline templates/generator.html.twig:17]: This style here doesn't seem to change anything below - checked on Firefox & chrome

File: templates/generator.html.twig [hunk 10]
```diff
@@ -30,9 +36,19 @@
...
-                <div class="col col-lg-6 col-md-6 note">
-                    <p>Note: if you want to use Phalcon, select "generic" application type, then look for Phalcon on the
-                        list of PHP extensions.</p>
+
+                <div class="row">
+                    <div class="col col-lg-6 col-md-6 note pl-30">
+                        <p>Note: if you want to use Phalcon, select "generic" application type, then look for Phalcon on the
+                            list of PHP extensions.</p>
+                    </div>
+
```
Comment (luispabon) [inline templates/generator.html.twig:45]: Please remove extra newline.

**Review by luispabon — CHANGES_REQUESTED** [review]
Just a few minor comments.

File: templates/generator.html.twig (hunk 9 above)
Comment (iGroovyboy) [reply templates/generator.html.twig:17]: It adds extra padding on the 'note' paragraph like [here ](https://imgur.com/a/DvlUIGa) to make it aligned on the left.

**Comment by iGroovyboy** [comment]
@luispabon fixed all the stuff in the comments except of style - it works on my side ([comment](https://github.com/phpdocker-io/phpdocker.io/pull/259#discussion_r665602253)), but if you still want it to be removed - let me know.

**Comment by luispabon** [comment]
Looking good, I'll look into that style separately. Thanks for your contribution!

**Pull request merged**